iree/turbine/kernel/wave/expansion/expansion.py

```diff
@@ -632,7 +632,8 @@ def fixup_reduction_nodes(
     expansion_context: ExpansionContext,
 ):
     reduction_context = expansion_context.reduction_context
-    for reduction in trace.walk(lambda x: isinstance(get_custom(x), Reduction)):
+    reduction_nodes = trace.walk(lambda x: isinstance(get_custom(x), Reduction))
+    for reduction in reversed(reduction_nodes):
```

Comment:
For education/documentation on github, can you give explain why we are reversing the order?

Reply:
In situations where we have multiple reductions, and the outputs of a
    reduction are used as inputs to another reduction, we need to ensure
    the fixup is done in the correct order, specifically from the last
    reduction to the first reduction since that is the order in
    which expansion proceeds. Also added a comment.